htdocs/core/class/commondocgenerator.class.php

```diff
@@ -587,7 +587,25 @@ public function get_substitutionarray_lines($line, $outputlangs)
 		$extralabels = $extrafields->fetch_name_optionals_label($extrafieldkey, true);
 		$line->fetch_optionals();
 
-		$resarray = $this->fill_substitutionarray_with_extrafields($line, $resarray, $extrafields, $array_key, $outputlangs);
+        $resarray = $this->fill_substitutionarray_with_extrafields($line, $resarray, $extrafields, $array_key, $outputlangs);
+        
+        // Check if the current line belongs to a supplier order
+        if (get_class($line) == 'CommandeFournisseurLigne') {
+            // Add the product supplier extrafields to the substitutions
+            $extralabels=$extrafields->fetch_name_optionals_label("product_fournisseur_price");
+            $columns = "";
+            foreach ($extralabels as $key => $value) {
+                $columns .= "$key, ";
+            }
+            $columns = substr($columns, 0, strlen($columns) - 2);
+            $resql = $this->db->query("SELECT $columns FROM " . MAIN_DB_PREFIX . "product_fournisseur_price_extrafields AS ex INNER JOIN " . MAIN_DB_PREFIX . "product_fournisseur_price AS f ON ex.fk_object = f.rowid WHERE f.ref_fourn = '" . $line->ref_supplier . "'");
```

Comment:
If there is no extrafields (the common case) ? What happened ? It seems this generates technical errors.